### DISCUSSION TITLE
Version 3.0.0

### DISCUSSION THREAD

**Discussion by bquorning** [rubocop-hq/rubocop-rspec#1880] [General]
The current `master` branch is being used for preparing the release of v3.0.0. There is a `2-x-stable` branch where we can backport bugfixes etc. that need to be released as a v2.x.x tag.

Please add relevant issues and pull requests to [the v3.0 milestone](https://github.com/rubocop/rubocop-rspec/milestone/2).

Feel free to add comments or replies to this discussion if that’s a better medium than opening a issue.

cc @rubocop/rubocop-rspec

**Comment by bquorning** [comment]
Should we enable _all_ pending cops, or _some_ pending cops?

**Reply by bquorning** [reply]
We talked a bit about this in #1245.

**Reply by bquorning** [reply]
I opened https://github.com/rubocop/rubocop-rspec/issues/1886

**Reply by ydah** [reply]
IMO, basically, I think there is no problem with enable. However, we should consider disabling it if there are many people who want to disable it due to Issues.

**Comment by pirj** [comment]
> It makes sense to synchronize the 3.0 release with the RuboCop 2.0 release

I withdraw this

> postponing the release to follow the parent RuboCop release cycle

and this, basing on the facts that:
1. Two years later, and the RuboCop 2.0 release is not any closer to become reality compared to back then
2. We have our reasons to release, the three extracted extensions to untangle from

**Comment by bquorning** [comment]
We are getting close. There is only [one issue](https://github.com/rubocop/rubocop-rspec/issues/1886) (with [accompanying PR](https://github.com/rubocop/rubocop-rspec/pull/1902)) left in [the 3.0 milestone](https://github.com/rubocop/rubocop-rspec/milestone/2).

I intend to merge that PR probably tomorrow, and make a pre-release (`v3.0.0.pre` ?), and then maybe next week we can do the proper `v3.0.0` release?

**Reply by pirj** [reply]
Fantastic!
Should we prepare a migration doc telling that users have to add extracted extensions manually?
What else? We’ve removed some node helpers?

**Reply by bquorning** [reply]
There is a migration doc at https://github.com/rubocop/rubocop-rspec/blob/master/docs/modules/ROOT/pages/upgrade_to_version_3.adoc, some of that should probably be mentioned in the changelog also (or at least we should have a link to the doc).

The removal of `RuboCop::RSpec::Language::NodePattern` is mentioned in the changelog. Do you think we need a more prominent mention? Did we remove other node helpers?

**Reply by pirj** [reply]
It seems to be solid to release then!

**Comment by bquorning** [comment]
It’s done! Thank you @ydah, @pirj, everyone ❤️
